Wikipedia/Code/WelcomeAnimationExtensions.swift

```diff
@@ -1,155 +1,54 @@
 import Foundation
 
-struct WelcomePlus {
-    var position: CGPoint
-    var animation: WelcomeAnimation
-}
-
-struct WelcomeCircle {
-    var position: CGPoint
-    var radius: CGFloat
-    var isDashed: Bool
-    var animation: WelcomeAnimation
-}
-
-struct WelcomeLine {
-    var position: CGPoint
-    var width: CGFloat
-    var animation: WelcomeAnimation
-}
-
-struct WelcomeImage {
-    var name: String
-    var frame: CGRect
-    var zPosition: CGFloat
-    var animation: WelcomeAnimation
-}
-
-struct WelcomeBar {
-    var frame: CGRect
-    var color: UIColor
-    var cornerRadius: CGFloat
-    var animation: WelcomeAnimation
+extension CGFloat {
+    func wmf_denormalizeUsingReference (reference: CGFloat) -> CGFloat {
```

Comment:
I think I know whats going on here… but not quite sure. Just hit me up after you get in.